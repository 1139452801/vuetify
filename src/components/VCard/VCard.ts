// Styles
import '../../stylus/components/_cards.styl'

// Components
import { VPaper } from '../VPaper'

// Mixins
import Routable from '../../mixins/routable'

// Helpers
import { addPaperClasses } from '../VPaper/VPaper'
import { convertToUnit } from '../../util/helpers'
import { deprecate } from '../../util/console'
import mixins from '../../util/mixins'

// Types
import { VNode } from 'vue'
import { ClassesObject } from './../../../types'

/* @vue/component */
export default mixins(
  Routable
).extend({
  name: 'v-card',

  data: () => ({
    isMouseOver: false as boolean
  }),

  props: {
    ...VPaper.options.props,
    elevation: {
      type: [Number, String],
      default: 1
    },
    flat: Boolean,
    height: [Number, String],
    hover: [Number, String],
    /* @deprecated */
    img: String,
    maxHeight: [Number, String],
    maxWidth: [Number, String],
    /* @deprecated */
    raised: Boolean,
    tile: Boolean,
    width: [Number, String]
  },

  computed: {
    classes (): ClassesObject {
      return {
        'v-paper v-card': true,
        'v-card--tile': this.tile
      }
    },
    computedElevation (): number | string {
      if (this.raised) {
        deprecate('<v-card raised>', '<v-card elevation="3">', this)

        return 3
      }

      if (this.hover === '' && this.isMouseOver) {
        deprecate('<v-card hover>', '<v-card hover="8">', this)

        return 8
      }

      return this.flat
        ? 0
        : this.isMouseOver ? this.hover : this.elevation
    },
    listeners (): object {
      return this.hover === undefined ? {} : {
        mouseenter: () => {
          this.isMouseOver = true
        },
        mouseleave: () => {
          this.isMouseOver = false
        }
      }
    },
    styles (): object {
      const style: Record<string, any> = {
        height: convertToUnit(this.height),
        maxHeight: convertToUnit(this.maxHeight),
        maxWidth: convertToUnit(this.maxWidth),
        width: convertToUnit(this.width)
      }

      if (this.img) {
        deprecate('<v-card img="...">', 'a nested <v-img>', this)

        style.background = `url("${this.img}") center center / cover no-repeat`
      }

      return style
    }
  },

  render (h): VNode {
    const { tag, data } = this.generateRouteLink()

    data.class = {
      ...data.class,
      ...addPaperClasses({
        color: this.color,
        dark: this.dark,
        light: this.light,
        elevation: this.computedElevation
      })
    }

    data.style = this.styles
    data.on = {
      ...data.on,
      ...this.listeners
    }

    return h(tag, data, this.$slots.default)
  }
})
