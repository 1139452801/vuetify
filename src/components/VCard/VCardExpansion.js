import { VExpandTransition , VSlideYReverseTransition } from '../transitions'

export default {
  name: 'v-card-expansion',

  props: {
    show: {
      type: Boolean,
      default: false
    },
    cover: {
      type: Boolean,
      default: false
    }
  },

  render (h) {
    const children = []

    children.push(h('div', {
      'class': 'v-card__text'
    }, this.$slots.default))
    const staticClass = 'v-card__expansion ' + (this.cover ? 'v-card__expansion--cover' : 'v-card__expansion--dropdown')
    const expansionStyle = h('div', {
      'class': staticClass,
      directives: [
        {
          name: 'show',
          value: this.show
        }
      ]
    }, children)

    return h(this.cover ? VSlideYReverseTransition : VExpandTransition, [expansionStyle])
  }
}
