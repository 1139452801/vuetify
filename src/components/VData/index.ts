import VLocalData from './VLocalData'
import VServerData from './VServerData'

export { VLocalData, VServerData }

export default {
  $_vuetify_subcomponents: {
    VLocalData,
    VServerData
  }
}
