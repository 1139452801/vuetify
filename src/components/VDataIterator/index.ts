import VDataIterator from './VDataIterator'
import VDataFooter from './VDataFooter'
import VDataIteratorServer from './VDataIteratorServer'

export { VDataIterator, VDataFooter, VDataIteratorServer }

export default {
  $_vuetify_subcomponents: {
    VDataIterator,
    VDataFooter,
    VDataIteratorServer
  }
}
