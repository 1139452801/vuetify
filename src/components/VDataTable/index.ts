import { createSimpleFunctional } from '../../util/helpers'

import VDataTable from './VDataTable'
import VDataHeader from './VDataHeader'
import VDataHeaderMobile from './VDataHeaderMobile'
import VRowGroup from './VRowGroup'
import VEditDialog from './VEditDialog'
import VRow from './VRow'
import VRowFunctional from './VRowFunctional'
import VRowRegular from './VRowRegular'
import VRowExpansion from './VRowExpansion'
import VTableVirtual from './VTableVirtual'

const VTableOverflow = createSimpleFunctional('v-table__overflow')

export {
  VDataTable,
  VEditDialog,
  VTableOverflow,
  VDataHeader,
  VDataHeaderMobile,
  VRowGroup,
  VRow,
  VRowRegular,
  VRowFunctional,
  VRowExpansion,
  VTableVirtual
}

export default {
  $_vuetify_subcomponents: {
    VDataTable,
    VEditDialog,
    VTableOverflow,
    VDataHeader,
    VDataHeaderMobile,
    VRowGroup,
    VRow,
    VRowRegular,
    VRowFunctional,
    VRowExpansion,
    VTableVirtual
  }
}
