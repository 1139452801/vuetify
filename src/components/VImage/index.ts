import VImage from './VImage'

/* istanbul ignore next */
VImage.install = function install (Vue) {
  Vue.component(VImage.options.name, VImage)
}

export { VImage }
export default VImage
