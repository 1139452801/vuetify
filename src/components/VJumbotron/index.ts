import VJumbotron from './VJumbotron'

/* istanbul ignore next */
VJumbotron.install = function install (Vue) {
  Vue.component(VJumbotron.options.name, VJumbotron)
}

export { VJumbotron }
export default VJumbotron
