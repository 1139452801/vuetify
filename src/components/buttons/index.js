import Btn from './Button'
import BtnToggle from './ButtonToggle.vue'
import SpeedDial from './SpeedDial'
import Morph from './Morph'

export default {
  Btn,
  BtnToggle,
  SpeedDial,
  Morph
}
