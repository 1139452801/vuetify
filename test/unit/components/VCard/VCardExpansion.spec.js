import { test } from '@/test'
import { VCardExpansion } from '@/components/VCard'

test('VCardExpansion.js', ({ mount }) => {
  it('should render with "display:none" applied and dropdown class', () => {
    const wrapper = mount(VCardExpansion)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with no "display:none" applied and dropdown class', () => {
    const wrapper = mount(VCardExpansion, {
      propsData: {
        show: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with "display:none" applied and cover class', () => {
    const wrapper = mount(VCardExpansion, {
      propsData: {
        cover: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with no "display:none" applied and cover class', () => {
    const wrapper = mount(VCardExpansion, {
      propsData: {
        cover: true,
        show: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
