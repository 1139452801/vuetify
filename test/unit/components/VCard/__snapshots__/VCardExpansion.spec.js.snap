// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VCardExpansion.js should render with "display:none" applied and cover class 1`] = `

<div class="v-card__expansion v-card__expansion--cover"
     style="display: none;"
>
  <div class="v-card__text">
  </div>
</div>

`;

exports[`VCardExpansion.js should render with "display:none" applied and dropdown class 1`] = `

<div class="v-card__expansion v-card__expansion--dropdown"
     style="display: none;"
>
  <div class="v-card__text">
  </div>
</div>

`;

exports[`VCardExpansion.js should render with no "display:none" applied and cover class 1`] = `

<div class="v-card__expansion v-card__expansion--cover">
  <div class="v-card__text">
  </div>
</div>

`;

exports[`VCardExpansion.js should render with no "display:none" applied and dropdown class 1`] = `

<div class="v-card__expansion v-card__expansion--dropdown">
  <div class="v-card__text">
  </div>
</div>

`;
